vsphere/resource_vsphere_host_virtual_switch.go

```diff
@@ -0,0 +1,135 @@
+	s["failback"].Default = true
+
+	s["allow_promiscuous"].Default = false
+	s["forged_transmits"].Default = true
+	s["mac_changes"].Default = true
+
+	s["shaping_enabled"].Default = false
+
+	return &schema.Resource{
+		Create: resourceVSphereHostVirtualSwitchCreate,
+		Read:   resourceVSphereHostVirtualSwitchRead,
+		Update: resourceVSphereHostVirtualSwitchUpdate,
+		Delete: resourceVSphereHostVirtualSwitchDelete,
+		Schema: s,
+	}
+}
+
+func resourceVSphereHostVirtualSwitchCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*govmomi.Client)
+	host := d.Get("host").(string)
+	datacenter := d.Get("datacenter").(string)
+	ns, err := hostNetworkSystemFromName(client, host, datacenter)
+	if err != nil {
+		return fmt.Errorf("error loading network system: %s", err)
+	}
+
+	timeout := time.Duration(float64(d.Timeout(schema.TimeoutCreate)) * 0.8)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	spec := expandHostVirtualSwitchSpec(d)
+	if err := ns.AddVirtualSwitch(ctx, d.Get("name").(string), spec); err != nil {
+		return fmt.Errorf("error adding host vSwitch: %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	return resourceVSphereHostVirtualSwitchRead(d, meta)
+}
+
+func resourceVSphereHostVirtualSwitchRead(d *schema.ResourceData, meta interface{}) error {
+	timeout := time.Duration(float64(d.Timeout(schema.TimeoutCreate)) * 0.8)
+	id := d.Id()
+	host := d.Get("host").(string)
+	datacenter := d.Get("datacenter").(string)
+	sw, err := hostVSwitchFromName(meta.(*govmomi.Client), id, host, datacenter, timeout)
+	if err != nil {
+		return fmt.Errorf("error fetching virtual switch data: %s", err)
+	}
+
+	if err := flattenHostVirtualSwitchSpec(d, &sw.Spec); err != nil {
+		return fmt.Errorf("error setting resource data: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVSphereHostVirtualSwitchUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*govmomi.Client)
+	host := d.Get("host").(string)
+	datacenter := d.Get("datacenter").(string)
+	ns, err := hostNetworkSystemFromName(client, host, datacenter)
+	if err != nil {
+		return fmt.Errorf("error loading network system: %s", err)
+	}
+
+	timeout := time.Duration(float64(d.Timeout(schema.TimeoutCreate)) * 0.8)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	spec := expandHostVirtualSwitchSpec(d)
+	if err := ns.UpdateVirtualSwitch(ctx, d.Id(), *spec); err != nil {
+		return fmt.Errorf("error updating host vSwitch: %s", err)
+	}
+
+	return resourceVSphereHostVirtualSwitchRead(d, meta)
+}
+
+func resourceVSphereHostVirtualSwitchDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*govmomi.Client)
+	ns, err := hostNetworkSystemFromName(client, d.Get("host").(string), d.Get("datacenter").(string))
+	if err != nil {
+		return fmt.Errorf("error loading network system: %s", err)
+	}
+
+	timeout := time.Duration(float64(d.Timeout(schema.TimeoutCreate)) * 0.8)
```

Comment:
as discussed on Slack - there's no guarantee a cancel would complete in `0.2 * timeout`?